packages/core/src/types.ts

```diff
@@ -27,30 +28,33 @@ export interface AnyEventObject extends EventObject {
   [key: string]: any;
 }
 
-/**
- * The full definition of an action, with a string `type` and an
- * `exec` implementation function.
- */
 export interface BaseActionObject {
-  /**
-   * The type of action that is executed.
-   */
   type: string;
-  [other: string]: any;
+  params?: Record<string, any>;
```

Comment:
The problem is that it's difficult to infer the types of these action params unless we strongly type those actions (e.g., with `createModel` or otherwise) - let's figure this out with typegen before putting `unknown` here